packages/nox/test/tpetra/Brusselator/Problem_Interface.hpp

```diff
@@ -0,0 +1,121 @@
+//
+//            NOX: An Object-Oriented Nonlinear Solver Package
+//                 Copyright (2002) Sandia Corporation
+//
+// Under terms of Contract DE-AC04-94AL85000, there is a non-exclusive
+// license for use of this work by or on behalf of the U.S. Government.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Roger Pawlowski (rppawlo@sandia.gov) or
+// Eric Phipps (etphipp@sandia.gov), Sandia National Laboratories.
+// ************************************************************************
+//  CVS Information
+//  $Source$
+//  $Author$
+//  $Date$
+//  $Revision$
+// ************************************************************************
+//@HEADER
+
+//-----------------------------------------------------------------------------
+#ifndef Tpetra_Problem_Interface_HPP
+#define Tpetra_Problem_Interface_HPP
+
+// Interface to the NOX_Epetra_Group to provide for
+// residual and matrix fill routines.
+
+// ---------- Standard Includes ----------
+#include <iostream>
+#include <Tpetra_Vector.hpp>
+#include <Tpetra_Operator.hpp>
+#include <Tpetra_RowMatrix.hpp>
+
+// ---------- Forward Declarations ----------
+template<typename ScalarType>
+class Brusselator;
+
+template <typename ScalarType>
+class Problem_Interface
+{
+  using ST = typename Tpetra::Vector<ScalarType>::scalar_type;
+  using LO = typename Tpetra::Vector<>::local_ordinal_type;
+  using GO = typename Tpetra::Vector<>::global_ordinal_type;
+ 
+  using tvector_t     = typename Tpetra::Vector<ST, LO, GO>;
```

Comment:
I'm doing this but it could be better to use `nox/src-tpetra/NOX_TpetraTypedefs.hpp`. However this namespace use `Tpetra::MultiVector<>::scalar_type` which doesn't have default enabled template types.

Reply:
@stmcgovern I think this approach makes sense (using our own aliasing instead of the provided #include file, to allow for templating). What do you think?